src/frontend/js/toggle-dashboard-section.ts

```diff
@@ -18,3 +21,91 @@ async function toggleDashboardSection({ target }: Event): Promise<void> {
   content.classList.toggle("max-h-160");
   content.classList.toggle("max-h-0");
 }
+
+
+async function loadRssFeed() {
+  const feedWidget = document.querySelector(".rss-feed");
+
+  if (feedWidget) {
+    const feedLoading = document.getElementById("feed-loading");
+    const reloadButton = document.getElementById("reload");
+    
+    feedLoading.classList.remove("hidden");
+    reloadButton.classList.add("hidden");
+    const url = feedWidget.attributes[1].value;
+
+    // set timeout
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => {
+      controller.abort();
+    }, 5000);
+
+    fetch(url, {signal: controller.signal}).then(async (response: Response) => {
+      if (response.status === 200) {
```

Comment:
I justed tested the error handling case e. g. with an invalid url like integreat-app.de/fed and noticed that we need two more things:
1. On a controller timeout we need similar functionality (replacing the loading icon by the reload button) like in the catch block to prevent an infinite loading spin
2. The catch block cannot capture 404 errors so I think something like `if !reponse.ok { throw new Error("some error")}` would be helpful: 
```
A fetch() promise will reject with a TypeError when a network error is encountered or CORS is misconfigured on the
server-side, although this usually means permission issues or similar — a 404 does not constitute a network error, for
example. An accurate check for a successful fetch() would include checking that the promise resolved, then checking
that the Response.ok property has a value of true. 
```

Reply:
Thanks for your feedback! I'm glad that the major part works fine. As far as I understand it, my changes should fix the two issues you pointed out.